dask/array/core.py

```diff
@@ -2853,13 +2856,17 @@ def atop(func, out_ind, *args, **kwargs):
                              % (ind, arg.ndim))
 
     numblocks = {a.name: a.numblocks for a, ind in arginds if ind is not None}
-    argindsstr = list(concat([(a if ind is None else a.name, ind) for a, ind in arginds]))
+    argindsstr = list(concat([(delayed(a) if sizeof(a) > 1e6 else a
+                               if ind is None else a.name, ind)
```

Comment:
I have to think about Python parsing rules to determine if this parses as

- `delayed(a) if sizeof(a) > 1e6 else (a if ind is None else a.name)`
-  `(delayed(a) if sizeof(a) > 1e6 else a) if ind is None else a.name`

I might put in the parens for how it's intended (even if unnecessary) just to make this clear on first glance.

Reply:
Fixed